### ISSUE TITLE
Ignore "create database" in mysql dump, if present

### ISSUE THREAD

**Issue by djavorszky** [djavorszky/ddnc#8]

**Comment by djavorszky** [comment]
Did a bit of research, you can't tell mysql to ignore the create database lines. You can only tell the mysqldump not to include them. Since that's not an option for us, I think we have following options:

Little background: It seems like the `mysqldump` command only includes the "CREATE DATABASE..." command if multiple databases have been exported into the same file. In that case, for each database `dbname`, the following lines are included:

> CREATE DATABASE /*!32312 IF NOT EXISTS*/ \`dbname\` /*!40100 DEFAULT CHARACTER SET utf8 */;
> 
> USE \`dbname\`;

After the above lines, `dbname` is no longer referenced in the file. With these in mind, we have the following options:

1. Replace `dbname` to what was specified in the request - this could be tricky, if for example the `dbname` corresponds with any column names, as that could cause issues. I'm sure we could regexp it though and a bit of magic (e.g. only check first n lines).
2. Fail fast - If there's a `create database` in there, that means there are more than one databases in the dumpfile. The question here is whether we want that to happen or not? The server will not know about the excess databases, which means the connector will import databases that noone will know about. In this case, it would probably make sense to fail the whole process and report back to the user that they'll have to do this manually, or remove the excess databases from there.

@sorin-pop, @zsoltbalogh, what do you think, which one should we go with?

Alternatively, as a future feature, we could make the connector realize that there are multiple connecting databases and report each of them back to the server, which could then handle them as a unit in the future. Dropping one would drop all, for example.

**Comment by djavorszky** [comment]
Little correction: 
> Little background: It seems like the mysqldump command only includes the "CREATE DATABASE..." command if multiple databases have been exported into the same file.

In reality, `mysqldump` includes this line even if only one database has been exported. For example:
`mysqldump --add-drop-table -u root -p -B dbname > dbname.sql`
Will include the create database line because of the `-B` flag. Its man page says:

>Dump several databases. Normally, mysqldump treats the first name argument on the command line as a database name and following names as table names. With this option, it treats all name arguments as database names. CREATE DATABASE and USE statements are included in the output before each new database.

**Comment by zsoltbalogh** [comment]
I think this is what we should do:
1) Check the file if it contains multiple CREATE DATABASE commands. If yes, fail.
2) If it only contains one, remove it from the dump and only import the other content. This is also true for the DROP DATABASE, USE, etc. commands. 
3) Importing should be done with the newly created users' privileges. This way the import should fail when there's something nasty in the database file.

**Comment by djavorszky** [comment]
Thanks @zsoltbalogh for your thoughts, I also think that's the best course of action. I'm working on implementing them.
Nr3 is already in as of e669520c425b5716ca4a820f90d8a90b01fc15f7, the others are en route

**Comment by djavorszky** [comment]
So I've implemented a couple of methods (in repository https://github.com/djavorszky/sutils), namely, one that searches in a case insensitive, and one that searches in a case sensitive way. Using a case insensitive search for "create database", it takes 2.5 seconds to go through a standard vanilla Liferay 6.2 mysql dump.

Doing the same with a case sensitive search, it takes 97 ms. Benchmarking also confirms this:
```
$ go test -bench Find
BenchmarkFindIgnoreCase-8          30000             54934 ns/op
BenchmarkFindCaseSensitive-8     1000000              2047 ns/op
```
(name of test, number of operations run, time per operation)

Given how the Case Sensitive search is 20 times faster, I'll use that to search for lowercase and uppercase occurrences each instead of do the ignore case search.

**Comment by zsoltbalogh** [comment]
MVP, Dani! :)

**Comment by djavorszky** [comment]
Implemented as of 9c5b71af5aed15b9cf21d758b15f7d7dbc4083ec
1. If there's only one `create`, `use` or `drop` statements each, they will be removed from the dump
2. If there's more than one `create`, `use` or `drop` statements each, import stops
